tasks.py

```diff
@@ -761,3 +798,132 @@ def frontend_build(c):
+            with open(dst.name, "wb") as f:
+                shutil.copyfileobj(response.raw, f)
+            print(f"Downloaded frontend build to temporary file: {dst.name}")
+
+            handle_extract(dst.name)
+
+    # if zip file is specified, try to extract it directly
+    if file:
+        handle_extract(file)
+        return
+
+    # check arguments
+    if ref is not None and tag is not None:
+        print("[ERROR] Do not set ref and tag.")
+        return
+
+    if ref is None and tag is None:
+        try:
+            ref = subprocess.check_output(["git", "rev-parse", "HEAD"], encoding="utf-8").strip()
+        except Exception:
+            print("[ERROR] Cannot get current ref via 'git rev-parse HEAD'")
+            return
+
+    if ref is None and tag is None:
+        print("[ERROR] Either ref or tag needs to be set.")
+
+    if tag:
+        tag = tag.lstrip("v")
+        try:
+            handle_download(f"https://github.com/{repo}/releases/download/{tag}/frontend-build.zip")
+        except Exception as e:
+            if not isinstance(e, requests.HTTPError):
+                raise e
+            print(f"""[ERROR] An Error occurred. Unable to download frontend build, release or build does not exist,
+try downloading the frontend-build.zip yourself via: https://github.com/{repo}/releases
+Then try continuing by running: invoke frontend-download --file <path-to-downloaded-zip-file>""")
+
+        return
+
+    if ref:
+        # get workflow run from all workflow runs on that particular ref
+        workflow_runs = requests.get(f"https://api.github.com/repos/{repo}/actions/runs?head_sha={ref}", headers=default_headers).json()
+
+        if not (qc_run := find_resource(workflow_runs["workflow_runs"], "name", "QC")):
+            print("[ERROR] Cannot find any workflow runs for current sha")
+            return
+        print(f"Found workflow {qc_run['name']} (run {qc_run['run_number']}-{qc_run['run_attempt']})")
+
+        # get frontend-build artifact from all artifacts available for this workflow run
+        artifacts = requests.get(qc_run["artifacts_url"], headers=default_headers).json()
+        if not (frontend_artifact := find_resource(artifacts["artifacts"], "name", "frontend-build")):
+            print("[ERROR] Cannot find frontend-build.zip attachment for current sha")
+            return
+        print(f"Found artifact {frontend_artifact['name']} with id {frontend_artifact['id']} ({frontend_artifact['size_in_bytes']/1e6:.2f}MB).")
+
+        print(f"""
+GitHub doesn't allow artifact downloads from anonymous users. Either download the following file
+via your signed in browser, or consider using a point release download via invoke frontend-download --tag <git-tag>
+
+    Download: https://github.com/{repo}/suites/{qc_run['check_suite_id']}/artifacts/{frontend_artifact['id']} manually and
+    continue by running: invoke frontend-download --extract --file <path-to-downloaded-zip-file>""")
```

Comment:
Good catch, I would remove the `--extract` option here, as it defaults to True and its also not used in all other suggestions. The reason why it was here is, that I first planned to use the extract option to also supply the path, but then quickly noticed that its not possible with a boolean setting, but I already had written this print statement.